server/protocols/jmap/src/main/java/org/apache/james/jmap/JMAPRoute.java

```diff
@@ -43,20 +43,21 @@
         public static class ReadyToBuild {
             private final Endpoint endpoint;
             private final Action action;
-            private boolean corsHeaders;
 
             ReadyToBuild(Endpoint endpoint, Action action) {
                 this.endpoint = endpoint;
                 this.action = action;
-                this.corsHeaders = false;
             }
 
-            public ReadyToBuild corsHeaders() {
-                this.corsHeaders = true;
-                return this;
+            public JMAPRoute corsHeaders() {
+                return build(true);
             }
 
-            public JMAPRoute build() {
+            public JMAPRoute noCorsHeaders() {
+                return build(false);
```

Comment:
passing a boolean here to have a `if` in the called method seems pointless, please pass the Action directly